### PR TITLE
Fix release year bug.

### PR DESCRIPTION
**Pull request by Somnibyte**
* Fixes release year bug.
* ComicFormViewController finish button now sends you back to the dashboard.

### CODE REVIEW

File: lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift [hunk 1]
```diff
@@ -236,6 +238,19 @@ final class ComicFormViewController: UIViewController, AlertDisplaying {
         releaseDateTextField.inputAccessoryView = releaseDateToolBar
     }
 
+    private func checkReleaseYear(completion: () -> Void) {
```
Comment (mayasaxena) [inline lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift]: This doesn't need to have a completion, you can just have it return a Bool and if it's false show the error there

File: lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift [hunk 2]
```diff
@@ -236,6 +238,19 @@ final class ComicFormViewController: UIViewController, AlertDisplaying {
...
+        guard
+            let releaseYearText = releaseDateTextField.text,
+            let releaseYear = Int(releaseYearText)
+            else { return }
+        if releaseYear < 2200 {
```
Comment (mayasaxena) [inline lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift]: 2200 should be saved as a constant in this class

**Review by mayasaxena — CHANGES_REQUESTED** [review]

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/073b18522e74d7732fd6cfe260bee171f410fda8/ComicFormViewController.swift#L354">ComicFormViewController.swift#L354</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/073b18522e74d7732fd6cfe260bee171f410fda8/lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift#L354">lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift#L354</a></strong>: immutable value ‘error’ was never used; consider replacing with ‘_’ or removing it  <br /><code>
                        case .failure(let error):
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.422 (0.424) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.543 (6.545) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-33-STQGZGTG3PQ6LB4TW4CWO326SH42K5CM2HUQEAOC5Y3ET5GUUEEA/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-33-STQGZGTG3PQ6LB4TW4CWO326SH42K5CM2HUQEAOC5Y3ET5GUUEEA/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_1PU7JZS316EO3-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.368 (0.370) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.517 (6.519) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift [hunk 3]
```diff
@@ -311,6 +332,41 @@ final class ComicFormViewController: UIViewController, AlertDisplaying {
...
+                self.present(alertController, animated: true, completion: nil)
+            }
+        }
+    }
+
+    private func dismissKeyboardAndNotifications() {
+        view.endEditing(true)
+        deregisterFromKeyboardNotifications()
+    }
+
```
Comment (ghost) [inline lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift:369]: MINOR Trailing Whitespace Violation: Lines should not have trailing whitespace. [rule](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

**Comment by ghost** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.431 (0.434) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 9.103 (9.105) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by mayasaxena — APPROVED** [review]

**Pull request merged**